src/Contexts/PrintLoadingContext.tsx

```diff
@@ -1,40 +1,85 @@
-import React, { useReducer } from 'react';
+import React, { useReducer, useEffect } from 'react';
 import logger from '../Utilities/Logger';
+import _ from 'lodash';
 
 type PrintLoadingProps = {
-    isDone: Promise<unknown>[],
     dispatch: React.Dispatch<PrintLoadingDispatch>
 };
 
-const PrintLoadingContext = React.createContext<Partial<PrintLoadingProps>>({isDone: []});
+const PrintLoadingContext = React.createContext<Partial<PrintLoadingProps>>({});
 
 type Props = {
   children: React.ReactNode;
 };
 
-type PrintLoadingDispatch = {type: PrintLoadingActions, payload: Promise<unknown>}
+type AddExpectedDispatch = {type: PrintLoadingActions.ADD_EXPECTED_PROMISE_COUNT, expected: number}
+type AddPromiseDispatch = {type: PrintLoadingActions.ADD_PROMISE | PrintLoadingActions.ADD_PROMISE_INCR_EXPECTED, payload: Promise<unknown>}
+type ResetExpectedDispatch = {type: PrintLoadingActions.RESET_EXPECTED_COUNT}
+type PrintDispatch = {type: PrintLoadingActions.PRINT, expected: number}
+type PrintLoadingDispatch = AddExpectedDispatch | AddPromiseDispatch | ResetExpectedDispatch | PrintDispatch;
+
+const print = _.debounce(window.print, 3000, {leading: false, trailing: true});
+
+interface PrintLoadingState {
+    expected: number;
+    arr: Promise<unknown>[];
+}
 
 export enum PrintLoadingActions {
-    ADD_PROMISE
+    ADD_PROMISE,
+    ADD_PROMISE_INCR_EXPECTED,
+    ADD_EXPECTED_PROMISE_COUNT,
+    RESET_EXPECTED_COUNT,
+    PRINT,
 }
 
-const reducer = (state: Promise<unknown>[], action: PrintLoadingDispatch): Promise<unknown>[] => {
-    logger.debug('Adding a promise to the Loading context');
+const reducer = (state: PrintLoadingState, action: PrintLoadingDispatch): PrintLoadingState => {
     switch(action.type) {
+    case PrintLoadingActions.RESET_EXPECTED_COUNT:
+        logger.debug('Printing state reset.');
```

Comment:
We might want to prepend some sort of tag, not a big deal though since I would like to be able to add that to the logger as awhole